shared/chat/conversation/attachment-input/index.js

```diff
@@ -1,76 +1,124 @@
 // @flow
-import React, {Component} from 'react'
+import * as React from 'react'
 import {Box, Button, Icon, Image, Input, PopupDialog, Text, ButtonBar} from '../../../common-adapters/index'
-import {globalColors, globalStyles, isMobile} from '../../../styles'
-import {isIPhoneX} from '../../../constants/platform'
-import type {Props} from '.'
+import {globalColors, globalStyles, isMobile, isIPhoneX} from '../../../styles'
 
+export type PathToInfo = {
+  [path: string]: {
+    type: 'image' | 'file',
+    title: string,
+    filename: string,
+  },
+}
+
+type Props = {
+  pathToInfo: PathToInfo,
+  onClose: () => void,
+  onSubmit: (pathToInfo: PathToInfo) => void,
+}
 type State = {
   index: number,
-  title: string,
+  pathToInfo: PathToInfo,
 }
 
-class RenderAttachmentInput extends Component<Props, State> {
+class RenderAttachmentInput extends React.Component<Props, State> {
   state: State
+  _input: ?Input
 
   constructor(props: Props) {
     super(props)
     this.state = {
       index: 0,
-      title: (props.inputs.length > 0 && props.inputs[0].title) || '',
+      pathToInfo: props.pathToInfo,
     }
   }
 
-  _onSelect = () => {
-    const close = this.state.index === this.props.inputs.length - 1
-    this.props.onSelect(this.props.inputs[this.state.index], this.state.title, close)
```

Comment:
before this would just keep track of the title and would mutate the parent value but instead we just keep track of the title in our own structure and tell the parent when we're done collecting the titles